spec/controllers/sessions_controller_spec.rb

```diff
@@ -1,5 +1,95 @@
-require 'rails_helper'
+require "rails_helper"
 
 RSpec.describe SessionsController, type: :controller do
+  describe "GET #new" do
+    let(:user) { create(:user) }
 
+    context "user has already logged in" do
+      before do
+        session[:user_id] = user.id
```

Comment:
Consider using a spec helper for logging in, since the mechanics for logging in a user may be more complex. So here you can just say `login_user` or just `login`.